experimental/CliffordAlgebras/src/CliffordAlgebras.jl

```diff
@@ -0,0 +1,579 @@
+  end
+
+  CliffordAlgebraElem(C::CliffordAlgebra{T,S}, coeff::Vector{T}) where {T,S} =
+    CliffordAlgebraElem{elem_type(C.base_ring),typeof(C.gram),typeof(C)}(C, coeff)
+
+  function CliffordAlgebraElem(C::CliffordAlgebra{T,S}, coeff::Vector{R}) where {T,S,R}
+    K = C.base_ring
+    @req __can_convert_coeff(coeff, K) "entries of coefficient vector are not contained in $(K)"
+    return CliffordAlgebraElem{elem_type(C.base_ring),typeof(C.gram),typeof(C)}(
+      C, K.(coeff)
+    )
+  end
+end
+
+elem_type(::Type{CliffordAlgebra{T,S}}) where {T,S} =
+  CliffordAlgebraElem{T,S,CliffordAlgebra{T,S}}
+
+parent_type(::Type{CliffordAlgebraElem{T,S,P}}) where {T,S,P} = P
+
+base_ring_type(C::CliffordAlgebra{T,S}) where {T,S} = typeof(base_ring(C))
+
+################################################################################
+#
+#  Construction
+#
+################################################################################
+
+### Algebra ###
+
+@doc raw"""
+    clifford_algebra(qs::QuadSpace) -> CliffordAlgebra
+
+Return the Clifford algebra of the quadratic space 'qs'.
+"""
+clifford_algebra(qs::Hecke.QuadSpace) =
+  CliffordAlgebra{elem_type(base_ring(qs)),typeof(gram_matrix(qs))}(qs)
+
+### Elements ###
+(C::CliffordAlgebra)() = CliffordAlgebraElem(C)
+
+function (C::CliffordAlgebra)(a::R) where {R<:RingElem}
+  res = fill(zero(a), dim(C))
+  res[1] = a
+  return CliffordAlgebraElem(C, res)
+end
+
+function (C::CliffordAlgebra)(a::R) where {R<:Number}
+  res = fill(zero(a), dim(C))
+  res[1] = a
+  return CliffordAlgebraElem(C, res)
+end
+
+(C::CliffordAlgebra)(coeff::Vector{R}) where {R} = CliffordAlgebraElem(C, coeff)
+
+################################################################################
+#
+#  Basic field access
+#
+################################################################################
+
+### Algebra ###
+@doc raw"""
+    base_ring(C::CliffordAlgebra) -> Ring
+
+Return the base ring of the Clifford algebra $C$.
+"""
+base_ring(C::CliffordAlgebra) = C.base_ring
+
+@doc raw"""
+    space(C::CliffordAlgebra) -> QuadSpace
+
+Return the underlying quadratic space of the Clifford algebra $C$.
+"""
+space(C::CliffordAlgebra{T,S}) where {T,S} = C.space::Hecke.QuadSpace{parent_type(T),S}
+
+@doc raw"""
+    gram_matrix(C::CliffordAlgebra) -> MatElem
+
+Return the Gram matrix of the free quadratic module with Clifford algebra $C$.
+"""
+gram_matrix(C::CliffordAlgebra) = C.gram
+
+@doc raw"""
+    dim(C::CliffordAlgebra) -> Int
+
+Return the dimension of the Clifford algebra $C$ over its base ring.
+"""
+dim(C::CliffordAlgebra) = C.dim
+
+### Elements ###
+@doc raw"""
+    parent(x::CliffordAlgebraElem) -> CliffordAlgebra
+
+Return the Clifford algebra containing $x$.
+"""
+parent(x::CliffordAlgebraElem) = x.parent
+
+@doc raw"""
+  coeff(x::CliffordAlgebraElem) -> Vector
+
+Return the coefficient vector of $x$ wrt the
+canonical basis of its parent Clifford algebra.  
+"""
+coeff(x::CliffordAlgebraElem) = x.coeffs
```

Comment:
Iirc, `coeff` usually takes an additional index and returns the corresponding coefficient. The function returning all coefficients is then called `coefficients`